servers/display_server.cpp

```diff
@@ -713,19 +713,23 @@ void DisplayServer::set_icon(const Ref<Image> &p_icon) {
 	WARN_PRINT("Icon not supported by this display server.");
 }
 
-DisplayServer::IndicatorID DisplayServer::create_status_indicator(const Ref<Image> &p_icon, const String &p_tooltip, const Callable &p_callback) {
+DisplayServer::IndicatorID DisplayServer::create_status_indicator(const Ref<Texture2D> &p_icon, const String &p_tooltip, const Callable &p_callback) {
 	WARN_PRINT("Status indicator not supported by this display server.");
 	return INVALID_INDICATOR_ID;
 }
 
-void DisplayServer::status_indicator_set_icon(IndicatorID p_id, const Ref<Image> &p_icon) {
+void DisplayServer::status_indicator_set_icon(IndicatorID p_id, const Ref<Texture2D> &p_icon) {
 	WARN_PRINT("Status indicator not supported by this display server.");
 }
 
 void DisplayServer::status_indicator_set_tooltip(IndicatorID p_id, const String &p_tooltip) {
 	WARN_PRINT("Status indicator not supported by this display server.");
 }
 
+void DisplayServer::status_indicator_set_menu(IndicatorID p_id, const RID &p_menu_rid) {
```

Comment:
The idea is to implement `NativeMenu` subclass for the DBus menu and use it to define menus:

```c++
NativeMenu *nm = NativeMenu::get_singleton();

// define menu:
RID menu_id = nm->create_menu();
nm->add_item(menu_id, "item 1", ....);
nm->add_item(menu_id, "item 2", ....);

// attach it to the indicator:
DisplayServer::get_singleton()->status_indicator_set_menu(ind_id, menu_id);
```

That indicator will get native handle/id and pass it to native API (set "org.freedesktop.StatusNotifierItem.Menu" to DBus object path fro the menu). And system will automatically handle showing it after this (will show it when necessary, send menu callback to `NativeMenu`, and indicator activation callback won't fire).

Reply:
And `PopupMenu`/`StatatusIncdicator` classes which are wrappers around `NativeMenu` and `DisaplyServer` APIs will handle it in the user-friendly manner, so you can simply make two nodes instead of using API directly.